src/pyramid/csrf.py

```diff
@@ -274,78 +278,92 @@ def check_csrf_origin(
     .. versionchanged:: 1.9
        Moved from :mod:`pyramid.session` to :mod:`pyramid.csrf`
 
+    .. versionchanged:: 2.0
+       Added the ``allow_no_origin`` option.
+
     """
 
     def _fail(reason):
         if raises:
-            raise BadCSRFOrigin(reason)
+            raise BadCSRFOrigin("Origin checking failed - " + reason)
         else:
             return False
 
-    if request.scheme == "https":
```

Comment:
Sorry you should probably view this file with `?w=1`, dedented this whole block.